script.js

```diff
@@ -1,4 +1,183 @@
-  return myOutputValue;
+// Version 2 Comfortable - (Mon - Thu: 2h - 3h)
+/*
+- Player input number of dice, number of player and then select game mode
+- Press start button to display the player and the player's optimal combination
+- The highest / Lowest combination wins
+*/
+const diceInput = document.querySelector("#dice-number");
+const playerInput = document.querySelector("#player-number");
+const gameModeInput = document.querySelector("#game-mode");
+const submitBtn = document.querySelector("#submit-button");
+const outputDiv = document.querySelector("#output-div");
+const resetBtn = document.querySelector("#reset-button");
+resetBtn.style.display = "none";
+
+let gameRecordArr = [];
+let playerCounter = 1;
+let diceRollMsg, winner, winnerMsg;
+let displayMsg = "";
+let numsArr = "";
+let gameBoard;
+
+// Toggle Buttons
+function toggleBtn() {
+  if (resetBtn.style.display == "none") {
+    submitBtn.style.display = "none";
+    resetBtn.style.display = "block";
+  } else {
+    submitBtn.style.display = "block";
+    resetBtn.style.display = "none";
+  }
+}
+
+// Submit button click event
+submitBtn.addEventListener("click", function () {
+  let result = main();
+  submitBtn.disable = true;
+  outputDiv.innerHTML = result;
+});
+
+// Reset button click event
+resetBtn.addEventListener("click", function () {
+  location.reload();
+});
+
+// Roll dice
+function rollDice() {
+  let diceNum = Math.floor(Math.random() * 6) + 1;
+  return diceNum;
+}
+
+// Bubble Sort: Array low --> high or high --> low
+function sortDiceNumArr(arr, mode) {
+  const swap = (arr, index1, index2) => {
+    [arr[index1], arr[index2]] = [arr[index2], arr[index1]];
+  };
+  for (let i = 0; i < arr.length; i++) {
+    for (let j = 0; j < arr.length - i; j++) {
+      if (arr[j] < arr[j + 1] && mode == "big") {
+        swap(arr, j, j + 1);
+      } else if (arr[j] > arr[j + 1] && mode == "small") {
+        swap(arr, j, j + 1);
+      }
+    }
+  }
+  return arr;
+}
+
+// Get the biggest/lowest dice number combination
+function getDiceCombo(num_of_dice, mode) {
+  let counter = 0;
+  let arr = [];
+  while (counter < Number(num_of_dice)) {
+    arr.push(rollDice());
+    counter += 1;
+  }
+  numsArr = arr.toString();
+  let sortedArr = sortDiceNumArr(arr, mode);
+  let numStr = "";
+  for (let i = 0; i < sortedArr.length; i++) {
+    numStr += String(sortedArr[i]);
+  }
+  let combination = Number(numStr);
+  return combination;
+}
+
+// Generate a sorted leaderboard to show each player's optimal number
+function generateLeaderboard(arr, mode) {
+  let record;
+  let showRecord = "";
+  let leaderboard = [];
+  for (let i = 0; i < arr.length; i++) {
+    let player = `Player ${i + 1}`;
+    leaderboard.push({ player });
+    leaderboard[i].value = arr[i];
+  }
+  if (mode == "big") {
+    record = leaderboard.sort((a, b) => (a.value < b.value ? 1 : -1));
+  } else {
+    record = leaderboard.sort((a, b) => (a.value > b.value ? 1 : -1));
+  }
```

Comment:
im seeing this now after typing the above lol, seems like you did implement a .sort with a comparison function here. you can probably do smth similar for the above

also this can just be (a, b) => b.value - a.value for line 97 and a.value - b.value for line 99